File: src/main/java/frc/robot/subsystems/fuelVision/FuelVisionIO.java
package frc.robot.subsystems.fuelVision;

import static edu.wpi.first.units.Units.Radians;

import org.littletonrobotics.junction.AutoLog;

import edu.wpi.first.units.measure.Angle;

public interface FuelVisionIO {
    
    public default void updateInputs(FuelVisionIOInputs inputs){}

    @AutoLog
    class FuelVisionIOInputs{
        public FuelObservation[] observations = {new FuelObservation(new RadialPos2d(0,0), new ScreenSize2d(0,0), 0.0, 0.0, 0.0)};
        public boolean connected = false;
    }
}
record FuelObservation(RadialPos2d screenPos, ScreenSize2d screensize, double area, double confidance, double dist){}
record RadialPos2d(double x, double y){}
record ScreenSize2d(double x, double y){}

File: src/main/kotlin/frc/robot/subsystems/fuelVision/FuelVisionIO.kt
package frc.robot.subsystems.fuelVision

import org.littletonrobotics.junction.AutoLog

interface FuelVisionIO {
    fun updateInputs(inputs: FuelVisionIOInputs) {}

    @AutoLog
    open class FuelVisionIOInputs {
        @JvmField var observations: Array<FuelObservation> = arrayOf(
            FuelObservation(RadialPos2d(0.0, 0.0), ScreenSize2d(0.0, 0.0), 0.0, 0.0, 0.0)
        )
        @JvmField var connected: Boolean = false
        @JvmField var timestamp: Double = 0.0
    }
}
